libslide/src/bignum/modulo.rs

```diff
@@ -0,0 +1,47 @@
+#![allow(clippy::suspicious_arithmetic_impl)]
+use crate::bignum::Bignum;
+use std::ops;
+
+impl ops::Rem for Bignum {
+    type Output = Bignum;
+
+    fn rem(self, rhs: Bignum) -> Bignum {
+        // we only support positive mod for now
+        if !rhs.dec.is_empty() {
+            panic!("The module fuction only supports whole numbers!");
+        }
+        let mut result = self;
+        let zero = Bignum::new("0".to_string()).unwrap();
+        while result >= zero {
+            result = result - rhs.clone();
+        }
```

Comment:
Can you make the implementation more efficient using binary skipping? First you subtract x, then 2x, then 4x, etc. If you go over you backup and start again from x. Lmk if this makes sense.

Also, only compare with `>`

Reply:
yeah I definitely can. never heard of that, but thanks for introducing it to me

Reply:
wait do you think it will be faster, because we need to them implement an extra multiply

Reply:
implement both and benchmark :-)